### ISSUE TITLE
Provider subf2m return no results because of a space at the end of the title in html

### ISSUE THREAD

**Issue by GabbeHags** [morpheus65535/bazarr#2616]
**Describe the bug**
When using the provider subf2m no results is found but there exist some on the website. It looks like this is because of a space in the title.

**To Reproduce**
1. Make a search with the provider subf2m
2. Expect to find results

**Expected behavior**
I expect that if there exist subtitles on the website they should be found by Bazarr.

**Screenshots**

**Software (please complete the following information):**
 - Bazarr: 1.4.3
 - Radarr version 5.8.3.8933
 - Sonarr version4.0.8.1874
 - OS: Windows-10-10.0.19045-SP0
 - Python 3.10.0

**Additional context**
I investigated some and found out that all the titles in the html contain a space at the end (see below) this result in that the regex `^(.+?)( \((\d{4})\))?$` and `^(.+?) [-\(]\s?(.*?) (season|series)\)?( \((\d{4})\))?$` does not find anything.

```
<li>
    <div class="title"> 
        <a href="/subtitles/civil-war">Civil War  (2024) </a>
    </div>
    <div class="subtle count"> 6 subtitles </div>
</li>
```

I see 2 solutions to this.
1. Change the regex 
    -  from `^(.+?)( \((\d{4})\))?$` to `^(.+?)( \((\d{4})\))?\s*$` 
    -  from `^(.+?) [-\(]\s?(.*?) (season|series)\)?( \((\d{4})\))?$` to `^(.+?) [-\(]\s?(.*?) (season|series)\)?( \((\d{4})\))?\s*$`
2. `.strip()` the `result.text` from `_gen_results(...)`

I tested solution 2 on my server and it found some subtitles.

**Comment by GabbeHags** [comment]
This is possible linked to: https://github.com/morpheus65535/bazarr/issues/2227

**Comment by dashtinejad** [comment]
@GabbeHags 
Very good finding. I think my issue #2617 is related to this issue as well. I think if we implement one of your two solutions, the issue should be resolved.

Would you mind creating a PR?

**Comment by morpheus65535** [comment]
I've just merged #2623. Thanks!
